verifiers/halo2/src/benchmarking.rs

```diff
@@ -0,0 +1,259 @@
+#![cfg(feature = "runtime-benchmarks")]
+
+use std::io::{BufReader, Read};
+
+use crate::Halo2;
+use frame_benchmarking::v2::*;
+use frame_support::traits::{Consideration, Footprint};
+use frame_system::RawOrigin;
+
+use pallet_aggregate::{funded_account, insert_domain};
+use pallet_verifiers::{Tickets, VkEntry, VkOrHash, Vks};
+use sp_core::U256;
+
+pub struct Pallet<T: Config>(crate::Pallet<T>);
+pub trait Config: crate::Config {}
+impl<T: crate::Config> Config for T {}
+pub type Call<T> = pallet_verifiers::Call<T, Halo2<T>>;
+
+fn init<T: pallet_aggregate::Config>() -> (T::AccountId, u32) {
+    let caller: T::AccountId = funded_account::<T>();
+    let domain_id = 1;
+    insert_domain::<T>(domain_id, caller.clone(), Some(1));
+    (caller, domain_id)
+}
+
+pub struct BenchData {
+    pub vk: Vec<u8>,
+    pub proof: Vec<u8>,
+    pub pubs: Vec<U256>,
+}
+
+pub fn valid_bench_data() -> BenchData {
+    let pubs_bytes = include_bytes!("resources/VALID_PUBS.bin").to_vec();
```

Comment:
How complex is this proof used for benchmarking ? Keep in mind that these weights will be used to compute fees and to determine how many extrinsics validators can put in a block. So they mustn't be too small in case of complex proofs neither too high in case of simple proofs.
Take a look, for instance, at the work we did for benching the Risc0 and Ultraplonk verifier to get an idea

Reply:
Added benchmark routines for more complex proofs (k=8,16,21) similar to ultraplonk benchmarks